pkg/cloudprovider/provider/gce/config.go

```diff
@@ -202,19 +202,24 @@ func newConfig(resolver *providerconfig.ConfigVarResolver, spec v1alpha1.Provide
 		return nil, fmt.Errorf("cannot retrieve preemptible: %v", err)
 	}
 
-	cfg.assignPublicIPAddress, err = resolver.GetConfigVarBoolValue(cpSpec.AssignPublicIPAddress)
-	if err != nil {
-		return nil, fmt.Errorf("failed to retrieve assignPublicIPAddress: %v", err)
+	// make it true by default
+	cfg.assignPublicIPAddress = true
+
+	if cpSpec.AssignPublicIPAddress != nil {
```

Comment:
Actually its way simpler to just set the default as you did by a `cfg.assignPublicIPAddress = true`, there is no reason to have AssignPublicIPAddress be a `*providerconfig.ConfigVarBool` instead of a literal

Reply:
isn't `GetConfigVarBoolValue()` would return `(false, error)` in case when there is no `assignPublicIPAddress` and effectively overwrite `cfg.assignPublicIPAddress` to false?

Reply:
Good point.

/lgtm
/approve